SubscriberPublMsgToMsgRetComplBean intermittent test failure

Seeing this test fail intermittently in the 3.x pipeline:

```
00:58:01,523 [ERROR] io.helidon.microprofile.messaging.inner.ChannelTest.innerChannelBeanTest(CdiTestCase)[33]  Time elapsed: 0.08 s  <<< FAILURE!
java.lang.AssertionError: 
Result doesn't match at io.helidon.microprofile.messaging.inner.subscriber.SubscriberPublMsgToMsgRetComplBean.assertValid(SubscriberPublMsgToMsgRetComplBean.java:61)
Expected: is <[TEst2, tESt3, teST1]>
     but: was <[TEst2, tESt3, teST1]>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at io.helidon.microprofile.messaging.AssertableTestBean.assertWithOrigin(AssertableTestBean.java:60)
	at io.helidon.microprofile.messaging.inner.subscriber.SubscriberPublMsgToMsgRetComplBean.assertValid(SubscriberPublMsgToMsgRetComplBean.java:61)
	at io.helidon.microprofile.messaging.inner.subscriber.SubscriberPublMsgToMsgRetComplBean$Proxy$_$$_WeldClientProxy.assertValid(Unknown Source)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at io.helidon.microprofile.messaging.inner.ChannelTest.innerChannelBeanTest(ChannelTest.java:68)
```

And saw it on 2.x:
```
[INFO] Running io.helidon.microprofile.messaging.inner.ChannelTest
Error:  Tests run: 105, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 9.279 s <<< FAILURE! - in io.helidon.microprofile.messaging.inner.ChannelTest
Error:  io.helidon.microprofile.messaging.inner.ChannelTest.innerChannelBeanTest(CdiTestCase)[41]  Time elapsed: 0.073 s  <<< FAILURE!
java.lang.AssertionError: 
Result doesn't match at io.helidon.microprofile.messaging.inner.subscriber.SubscriberPublMsgToMsgRetComplBean.assertValid(SubscriberPublMsgToMsgRetComplBean.java:62)
Expected: is <[TEst2, tESt3, teST1]>
     but: was <[]>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at io.helidon.microprofile.messaging.AssertableTestBean.assertWithOrigin(AssertableTestBean.java:60)
	at io.helidon.microprofile.messaging.inner.subscriber.SubscriberPublMsgToMsgRetComplBean.assertValid(SubscriberPublMsgToMsgRetComplBean.java:62)
	at io.helidon.microprofile.messaging.inner.subscriber.SubscriberPublMsgToMsgRetComplBean$Proxy$_$$_WeldClientProxy.assertValid(Unknown Source)
```

--- a/microprofile/messaging/core/src/test/java/io/helidon/microprofile/messaging/inner/subscriber/SubscriberPublMsgToMsgRetComplBean.java
+++ b/microprofile/messaging/core/src/test/java/io/helidon/microprofile/messaging/inner/subscriber/SubscriberPublMsgToMsgRetComplBean.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, 2023 Oracle and/or its affiliates.
+ * Copyright (c) 2020 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -19,6 +19,7 @@ package io.helidon.microprofile.messaging.inner.subscriber;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
 import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.ExecutorService;
 
 import javax.enterprise.context.ApplicationScoped;
@@ -42,6 +43,7 @@ import org.reactivestreams.Publisher;
 public class SubscriberPublMsgToMsgRetComplBean implements AssertableTestBean, AsyncTestBean {
 
     CopyOnWriteArraySet<String> resultData = new CopyOnWriteArraySet<>();
+    private final CountDownLatch countDownLatch = new CountDownLatch(TEST_DATA.size());
 
     private final ExecutorService executor = createExecutor();
 
@@ -54,11 +56,15 @@ public class SubscriberPublMsgToMsgRetComplBean implements AssertableTestBean, A
 
     @Incoming("cs-void-message")
     public CompletionStage<Void> consumeMessageAndReturnCompletionStageOfVoid(Message<String> message) {
-        return CompletableFuture.runAsync(() -> resultData.add(message.getPayload()), executor);
+        return CompletableFuture.runAsync(() -> {
+            resultData.add(message.getPayload());
+            countDownLatch.countDown();
+        }, executor);
     }
 
     @Override
     public void assertValid() {
+        await("Messages not delivered in time!", countDownLatch);
         assertWithOrigin("Result doesn't match", resultData, is(TEST_DATA));
     }
 

--- a/microprofile/messaging/core/src/test/java/io/helidon/microprofile/messaging/inner/subscriber/SubscriberPublMsgToMsgRetComplBean.java
+++ b/microprofile/messaging/core/src/test/java/io/helidon/microprofile/messaging/inner/subscriber/SubscriberPublMsgToMsgRetComplBean.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020 Oracle and/or its affiliates.
+ * Copyright (c) 2020, 2024 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/microprofile/messaging/core/src/test/java/io/helidon/microprofile/messaging/inner/subscriber/SubscriberPublMsgToPaylRetComplVoidBean.java
+++ b/microprofile/messaging/core/src/test/java/io/helidon/microprofile/messaging/inner/subscriber/SubscriberPublMsgToPaylRetComplVoidBean.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, 2023 Oracle and/or its affiliates.
+ * Copyright (c) 2020 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -19,6 +19,7 @@ package io.helidon.microprofile.messaging.inner.subscriber;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.CompletionStage;
 import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.ExecutorService;
 
 import javax.enterprise.context.ApplicationScoped;
@@ -42,6 +43,7 @@ import org.reactivestreams.Publisher;
 public class SubscriberPublMsgToPaylRetComplVoidBean implements AssertableTestBean, AsyncTestBean {
 
     CopyOnWriteArraySet<String> resultData = new CopyOnWriteArraySet<>();
+    private final CountDownLatch countDownLatch = new CountDownLatch(TEST_DATA.size());
     private final ExecutorService executor = createExecutor();
 
     @Outgoing("cs-void-payload")
@@ -53,11 +55,15 @@ public class SubscriberPublMsgToPaylRetComplVoidBean implements AssertableTestBe
 
     @Incoming("cs-void-payload")
     public CompletionStage<Void> consumePayloadAndReturnCompletionStageOfVoid(String payload) {
-        return CompletableFuture.runAsync(() -> resultData.add(payload), executor);
+        return CompletableFuture.runAsync(() -> {
+            resultData.add(payload);
+            countDownLatch.countDown();
+        }, executor);
     }
 
     @Override
     public void assertValid() {
+        await("Messages not delivered in time!", countDownLatch);
         assertWithOrigin("Result doesn't match", resultData, is(TEST_DATA));
     }
 

--- a/microprofile/messaging/core/src/test/java/io/helidon/microprofile/messaging/inner/subscriber/SubscriberPublMsgToPaylRetComplVoidBean.java
+++ b/microprofile/messaging/core/src/test/java/io/helidon/microprofile/messaging/inner/subscriber/SubscriberPublMsgToPaylRetComplVoidBean.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020 Oracle and/or its affiliates.
+ * Copyright (c) 2020, 2024 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

